src/mrack/providers/aws.py

```diff
@@ -209,9 +209,47 @@ async def validate_hosts(self, reqs):
         """Validate that host requirements are well specified."""
         return
 
+    def get_subnet_available_ips(self, subnet_id, log_msg_start):
+        """Get number of IPs available in a subnet."""
+        try:
+            subnet = self.ec2.Subnet(subnet_id)
+        except ClientError:
+            err_msg = (
+                f"{log_msg_start} Error retrieving " f"info from subnet: {subnet_id}"
+            )
+            logger.error(err_msg)
+            raise
+
+        return subnet.available_ip_address_count
+
     async def can_provision(self, hosts):
-        """Check that hosts can be provisioned."""
-        return True
+        """Check that all host can be provisioned.
+
+        Checks:
+        * Available IPv4 addresses are enough
+        """
+        log_msg_start = self.dsp_name
+
+        available_ips = 0
+        # Get subnets ids from first subnet (they're the same for every host)
+        if subnet_ids := hosts[0].get("subnet_ids"):
+            for subnet_id in subnet_ids:
+                available_ips += self.get_subnet_available_ips(subnet_id, log_msg_start)
+        elif subnet_id := hosts[0].get("subnet_id"):
+            available_ips += self.get_subnet_available_ips(subnet_id, log_msg_start)
+        else:
+            logger.warning(
+                f"{log_msg_start} No subnet/s specified. "
+                "Could not check IP addresses availability."
+            )
+            return True
```

Comment:
Default behaviour when no subnet is specified is ignore it and go forward (same as before).
What are your thougths on this?

Reply:
If we do not have such information in the provisioning config this behavior of continue as is is a good move IMO maybe it does not even have to be warning but just info.

There are few concerns I have about the specific details. Lets meet I will present them to you.

I think we should do similar logic later as in OpenStack which open us the door to the network picking methods or functions. We should have the information about the networks in the provisioning config right?

So in general, you should extract "from config hierarchy" the subnet_id/subnet_ids and then just use the subnet or if multiple specified pick one (in case there is list in provisioning config and host do not specify the network).
However this part should be in different method bound to the create_server()

Reply:
Comments addressed.
We can now specify subnet/s in provider config (metadata) as well as in provisioning config.
The reason for keeping "subnet_id" is to support old metadata files, but it'll be legacy and "subnet_ids" should be used.

The priority order is: 
subnet_ids in provider config > subnet_id in provider config > subnet_ids in provisioning config > subnet_id in provisioning config

Reply:
@Tiboris sorry, I accidentally deleted your comment in this thread

Reply:
No worries